src/metal.cs

```diff
@@ -2050,5 +2399,242 @@ interface MTLStageInputOutputDescriptor : NSCopying
 		[Export ("reset")]
 		void Reset ();
 	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(NSObject))]
+	interface MTLType
+	{
+		[Export ("dataType")]
+		MTLDataType DataType { get; }
+	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(NSObject))]
+	interface MTLPointerType
+	{
+		[Export ("elementType")]
+		MTLDataType ElementType { get; }
+
+		[Export ("access")]
+		MTLArgumentAccess Access { get; }
+
+		[Export ("alignment")]
+		nuint Alignment { get; }
+
+		[Export ("dataSize")]
+		nuint DataSize { get; }
+
+		[Export ("elementIsArgumentBuffer")]
+		bool ElementIsArgumentBuffer { get; }
+
+		[NullAllowed, Export ("elementStructType")]
+		MTLStructType ElementStructType { get; }
+
+		[NullAllowed, Export ("elementArrayType")]
+		MTLArrayType ElementArrayType { get; }
+	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(MTLType))]
+	interface MTLTextureReferenceType
+	{
+		[Export ("textureDataType")]
+		MTLDataType TextureDataType { get; }
+
+		[Export ("textureType")]
+		MTLTextureType TextureType { get; }
+
+		[Export ("access")]
+		MTLArgumentAccess Access { get; }
+
+		[Export ("isDepthTexture")]
+		bool IsDepthTexture { get; }
+	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	interface IMTLCaptureScope { }
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[Protocol, Model]
+	[BaseType (typeof(NSObject))]
+	interface MTLCaptureScope
+	{
+		[Abstract]
+		[Export ("beginScope")]
+		void BeginScope ();
+
+		[Abstract]
+		[Export ("endScope")]
+		void EndScope ();
+
+		[Abstract]
+		[NullAllowed, Export ("label")]
+		string Label { get; set; }
+
+		[Abstract]
+		[Export ("device")]
+		IMTLDevice Device { get; }
+
+		[Abstract]
+		[NullAllowed, Export ("commandQueue")]
+		IMTLCommandQueue CommandQueue { get; }
+	}
+
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface MTLCaptureManager
+	{
+		[Static]
+		[Export ("sharedCaptureManager")]
+		MTLCaptureManager SharedCaptureManager { get; }
```

Comment:
```
MTLCaptureManager. SharedCaptureManager.X ();
```
is a mouthful, let's do like swift [1]
```
MTLCaptureManager. Shared.X ();
```

[1] https://developer.apple.com/documentation/metal/mtlcapturemanager/2869720-shared